src/jit/optimizer.cpp

```diff
@@ -3202,46 +3205,22 @@ void Compiler::optUnrollLoops()
                 }
             }
 
-            /* Finish the linked list */
-
-            if (loopList)
+            // Gut the old loop body
+            for (block = head->bbNext;; block = block->bbNext)
             {
-                loopList->gtPrev = loopLast;
-                loopLast->gtNext = nullptr;
-            }
-
-            /* Replace the body with the unrolled one */
-
-            block = head;
-
-            do
-            {
-                block = block->bbNext;
-                noway_assert(block);
                 block->bbTreeList = nullptr;
                 block->bbJumpKind = BBJ_NONE;
-                block->bbFlags &= ~BBF_NEEDS_GCPOLL;
-            } while (block != bottom);
-
-            bottom->bbJumpKind = BBJ_NONE;
-            bottom->bbTreeList = loopList;
-            bottom->bbFlags &= ~BBF_NEEDS_GCPOLL;
-            bottom->modifyBBWeight(bottom->bbWeight / BB_LOOP_WEIGHT);
-
-            bool dummy;
-
-            fgMorphStmts(bottom, &dummy, &dummy, &dummy);
-
-            /* Update bbRefs and bbPreds */
-            /* Here head->bbNext is bottom !!! - Replace it */
-
-            fgRemoveRefPred(head->bbNext, bottom);
-
-            /* Now change the initialization statement in the HEAD to "lvar = lval;"
-             * (the last value of the iterator in the loop)
-             * and drop the jump condition since the unrolled loop will always execute */
+                block->bbFlags &= ~(BBF_NEEDS_GCPOLL | BBF_LOOP_HEAD);
+                if (BasicBlock* jumpDest = block->bbJumpDest)
```

Comment:
If this cannot be if ((BasicBlock* jumpDest = block->bbJumpDest) != nullptr), please move the declaration out of the condition s.t. the condition can test against nullptr.